homeassistant/components/light/xiaomi_miio.py

```diff
@@ -236,9 +258,21 @@ def async_update(self):
             _LOGGER.debug("Got new state: %s", state)
 
             self._state = state.is_on
-            self._brightness = ceil((255/100.0) * state.brightness)
+            self._brightness = ceil((255 / 100.0) * state.brightness)
+
+            if state.delay_off_countdown > 0:
+                delayed_turn_off = dt.utcnow().replace(microsecond=0) + \
```

Comment:
You're not allowed to store "time left" or "time progressed" in the state attributes. You are only allowed to store timestamps and the time it was setup with. In this case, let's remove it.

Reply:
To my understanding this would be a "when to turn it off" timestamp. In that case it should simply track if there has been changes to it and adjust accordingly, not update on every `update()`. Would that work?

An automation / w/e for this usecase would be to calculate the delta to this timestamp to see how much time is still left until it gets turned off.

Reply:
@rytilahti Your undestanding is correct. I will try to avoid updating the timestamp on every `update()` by prediction + window.